### PR TITLE
ddl: directly use BackendConfig rather than use lightning config

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/54436

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/lightning/config/config.go [hunk 1]
```diff
@@ -1636,6 +1636,7 @@ func (cfg *Config) Adjust(ctx context.Context) error {
 
 // AdjustForDDL acts like Adjust, but DDL will not use some functionalities so
 // those members are skipped in adjusting.
+// TODO(lance6716): delete it
```
Comment (D3Hunter) [inline pkg/lightning/config/config.go:1639]: after this pr we can move this `pkg` to `lightning` dir

**Review by D3Hunter — APPROVED** [review]

**Review by tangenta — APPROVED** [review]

**Pull request merged**